File: 3442. Maximum Difference Between Even and Odd Frequency I/sol.c
int maxDifference(char* s) {
    int min_even = 100, max_odd = 0, counter[26];
    int index = 0;
    while (s[index] != '\0') {
        counter[s[index] - 'a']++;
        index++;
    }
    for (int i = 0; i < 26; i++) {
        if (counter[i] % 2 == 0) {
            if (counter[i] != 0)
                min_even = min_even > counter[i] ? counter[i] : min_even;
        } else
            max_odd = max_odd > counter[i] ? max_odd : counter[i];
    }

    return max_odd - min_even;
}

File: 3354. Make Array Elements Equal to Zero/sol.go
package main

func countValidSelections(nums []int) int {
	left := make([]int, len(nums))
	right := make([]int, len(nums))
	counter := 0
	for i, val := range nums {
		counter += val
		left[i] = counter
	}

	counter = 0
	for i := len(nums) - 1; i > -1; i-- {
		counter += nums[i]
		right[i] = counter
	}
	count := 0

	for i := range nums {
		if nums[i] == 0 {
			prev := getWithDefault(left, i-1, 0)
			next := getWithDefault(right, i+1, 0)
			if prev == next {
				count += 2
			}
			if abs(prev-next) == 1 {
				count++
			}
		}
	}

	return count
}

func getWithDefault(arr []int, index, def int) int {
	if index < 0 || index >= len(arr) {
		return def
	}
	return arr[index]
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
